amsd/models.py

```diff
@@ -117,7 +117,29 @@ class MessageStick(CustomModelMixin, Contribution):
     irn = Column(Unicode)
     notes = Column(Unicode)
     data_entry = Column(Unicode)
-    linked_filenames = Column(Unicode)
 
-    def get_linked_filenames(self, pks, image_type='thumbnail'):
-        return get_linked_filename_urls(pks, image_type)
+    def get_images(self, image_type='thumbnail', width='40'):
+        if not self.files:
+            return ''
+        res = []
+        cdstar_url = 'https://cdstar.shh.mpg.de/bitstreams/'
+        for k, f in self.files.items():
+            if image_type in ['web', 'thumbnail']:
+                res.append(
+                        HTML.a(
+                            HTML.img(
+                                alt = f.name,
+                                title = f.name,
+                                width = '%spx' % (width) if width else 'auto',
+                                src = '%s/%s' % (cdstar_url, f.jsondata.get(image_type)),
+                            ),
+                            href = '%s/%s' % (cdstar_url, f.jsondata.get('url')),
+                            target = '_new')
+                        )
+            else:
+                res.append(HTML.img(
+                    alt = f.name,
+                    title = f.name,
+                    src = '%s/%s' % (cdstar_url, f.jsondata.get('url')))
+                )
+        return '&nbsp;'.join(res)
```

Comment:
Most of this functionality is already available from [clldmpg.cdstar](https://github.com/clld/clldmpg/blob/master/src/clldmpg/cdstar.py).